flujo/state/backends/postgres.py

```
@@ -109,6 +113,13 @@ async def _verify_schema(self, pool: Pool) -> None:
 
     async def _init_schema(self, pool: Pool) -> None:
         async with pool.acquire() as conn:
+            # Serialize migrations across processes/containers using a Postgres advisory lock.
+            # This prevents migration stampedes on scale-up.
+            try:
+                await conn.execute("SELECT pg_advisory_xact_lock($1)", MIGRATION_LOCK_KEY)
+            except Exception:
+                # If advisory lock fails (e.g., permissions), continue without it to maintain backward compatibility.
+                pass
```

Ruff (0.14.7)

`try`-`except`-`pass` detected, consider logging the exception

(S110)